Description: Detect if upstart or systemd are running, not if their corresponding configfiles are installed
 Also change the systemctl path to fit Debian's
Author: Didier Raboud <odyx@debian.org>
Origin: vendor
Bug-Debian: http://bugs.debian.org/725394
Last-Update: 2013-10-07
--- a/usb_modeswitch.sh
+++ b/usb_modeswitch.sh
@@ -80,9 +80,9 @@
 PATH=/bin:/sbin:/usr/bin:/usr/sbin
 		. /lib/udev/hotplug.functions
 		wait_for_file /usr/sbin/usb_modeswitch_dispatcher
-		if [ -e "/etc/systemd/system/usb_modeswitch@.service" ]; then
+		if [ -d "/run/systemd/system/" ]; then # Test if systemd is running
 			exec systemctl --no-block start usb_modeswitch@$p1'_'$p2.service
-		elif [ -e "/etc/init/usb-modeswitch-upstart.conf" ]; then
+ 		elif [ -x /sbin/initctl ] && /sbin/initctl version 2>/dev/null | /bin/grep -q upstart; then # Test if upstart is running
 			exec initctl emit --no-wait usb-modeswitch-upstart UMS_PARAM=$1
 		else
 			exec usb_modeswitch_dispatcher --switch-mode $1 &
